### DISCUSSION TITLE
Bundle Identifier is variable in Info.plist

### DISCUSSION THREAD

**Discussion by nicop2000** [fastlane/fastlane#21858] [Q&A]
when I use get_ipa_info_plist_value I get $(APP_BUNDLE_ID) back because it says so in my Info.plist. But Xcode interprets it correctly as the variable it ist. Fastlane doesn't. Any tips on how to resolve this?
Because of different configurations (XConfig) I can't change it in the Info.plist file

**Comment by florence0239** [comment]
Hello,

To resolve the issue with Fastlane not interpreting $(APP_BUNDLE_ID) correctly in your Info.plist, try these steps:

Set Environment Variable
` ENV['APP_BUNDLE_ID'] = 'com.yourcompany.yourapp'`

Use get_info_plist_value:
`bundle_id = get_info_plist_value(
  path: "path/to/Info.plist",
  key: "CFBundleIdentifier"
)`

Preprocess Info.plist:
`/usr/libexec/PlistBuddy -c "Set :CFBundleIdentifier com.yourcompany.yourapp" "path/to/Info.plist"

`
These steps should help Fastlane correctly interpret the variable. 

Best Regard,
florence0239
[hpinstantink](https://www.hpinstantink.com.co)
